main/src/main/java/org/apache/karaf/main/Main.java

```diff
@@ -367,7 +362,37 @@ private void doMonitor() throws Exception {
     Lock getLock() {
         return lock;
     }
-    
+
+    private void watchdog() {
+        if(Boolean.getBoolean("karaf.systemd.enabled")) {
+            new Thread("Karaf Systemd Watchdog Thread") {
```

Comment:
I would add the `karaf.systemd.enabled` property (commented) in `etc/config.properties` to document it for users.